src/gui/src/SampleEditor/SampleEditor.cpp

```diff
@@ -931,11 +976,38 @@ void SampleEditor::valueChangedProcessingTypeComboBox( const QString unused )
 		default:
 			__loops.mode = Sample::Loops::FORWARD;
 	}
-	setUnclean();
+	if (previous != __loops.mode) {
+		setUnclean();
+	}
+	doneEditing();
 }
 
 
 
+
+void SampleEditor::valueChangedEditTypeComboBox( int index )
+{
+	if ( ! m_pTargetSampleView ) {
+		// qWarning() << "no TargetSampleView!";
+		return;
+	}
+	switch ( index ){
+		case 0 ://
+			m_pTargetSampleView->setEditMode( SampleEditor::EnvelopeType::VelocityEnvelope );
+			break;
+		case 1 ://
+			if ( m_pTargetSampleView ) {
```

Comment:
This sanity check and the one below should already covered by the one in the beginning of the function